.github/workflows/connector-test-command.yml

```diff
@@ -0,0 +1,315 @@
+          if [ -z "$CONNECTORS" ]; then
+            echo "No connectors modified. Setting matrix to empty string."
+            echo "matrix=" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+          # quote each comma‑separated connector and rebuild the list in one step
+          json_elements="\"${CONNECTORS//,/\",\"}\""
+          json_output="{ \"connector\": [${json_elements}] }"
+          echo "matrix=$json_output" | tee -a "$GITHUB_OUTPUT"
+
+    outputs:
+      connector-matrix: ${{ steps.convert-matrix.outputs.matrix }}
+
+  connectors-test-matrix:
+    name: Test Connectors
+    needs: [generate-matrix]
+    runs-on: ubuntu-24.04
+    env:
+      GCP_GSM_CREDENTIALS: ${{ secrets.GCP_GSM_CREDENTIALS }}
+    if: ${{ needs.generate-matrix.outputs.connector-matrix != '' }}
+    strategy:
+      matrix: ${{ fromJson(needs.generate-matrix.outputs.connector-matrix) }}
+      max-parallel: 5  # Limit number of parallel jobs
+      fail-fast: false  # Don't stop on first failure
+    steps:
+      - name: Checkout Airbyte
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      # Java deps
+      - uses: actions/setup-java@v4
+        with:
+          distribution: zulu
+          java-version: 21
+          cache: gradle
+
+      - uses: gradle/actions/setup-gradle@v3
+        with:
+          cache-read-only: false
+          cache-write-only: false
+
+      # Python deps
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+          check-latest: true
+          update-environment: true
+      - name: Install and configure Poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.8.5
+
+      - name: Install the latest version of uv
+        uses: astral-sh/setup-uv@v6
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+            build-essential \
+            libssl-dev \
+            libffi-dev \
+            python3-dev
+
+      - name: Install Poe and Dev Dependencies
+        run: |
+          # Install Poe so we can run the connector tasks:
+          uv tool install poethepoet
+
+          # These should be included as dev dependencies in the connector, but
+          # for many connectors they are undeclared:
+          uv tool install ruff
+          uv tool install pytest
+          uv tool install mypy
+
+      - name: Install connector dependencies
+        run: |
+          poe connector ${{ matrix.connector }} install
+
+      - name: Fetch connector secrets
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install pipx
+          pipx run airbyte-cdk[dev] secrets fetch \
+            --connector-name=${{ matrix.connector }} \
+            --print-ci-secrets-masks
+
+      - name: Run 'test-all' Poe Command
+        run: |
+          poe connector ${{ matrix.connector }} test-all
```

Comment:
Hmm, don't we need a build command here as well/first so the docker images are built, and are able to run integration tests on them?

Reply:
Follow up: I think this is where I'd insert my new build step in.

Reply:
@davinchia - The tests are run from the GitHub Actions worker runner. As of now, they run without needing an build step. You could add the image build step here, but it isn't strictly needed.

My plan (as of now) is still to create another "Docker Image Smoke Test" workflow similar to what is already here in the PR, but just running a simple `spec` and `check` on the image, rather than re-running _all_ tests on the image.

Reply:
@aaronsteers I might be misunderstanding, I don't think this support the current Java connectors workflow, where we also want to run Docker tests in addition to unit tests.

Reply:
@davinchia - We should try to confirm... and maybe I've misunderstood something along the way.

In this PR, I think the only Java source we had a dummy-change for was `source-mysql`, and it completed successfully all of the `gradle check` actions for the connector. Logs here: https://github.com/airbytehq/airbyte/actions/runs/14848712682/job/41688134425?pr=58680

And summary report [here](https://github.com/airbytehq/airbyte/actions/runs/14848712682/attempts/1#summary-41688134425):

<img width="790" alt="image" src="https://github.com/user-attachments/assets/a3e2b983-8671-4a13-8fb1-bb39dea3ebc6" />

Reply:
I've added a dummy change to trigger the pipeline also for `source-postgres`. Lmk in the meanwhile if you see anything "off" in the above logs, or if there's another action besides `gradle check` that we need to run.

Reply:
@aaronsteers MYSQL might be the odd one out. If you look at Source-MSSQL or Source-Mongodb-V2, you'll see 
an `test-integration` folder alongside the `test` folder. These are typically tests that invoke Docker. The same pattern exists on Destinations.

<img width="434" alt="Screenshot 2025-05-05 at 5 22 18 PM" src="https://github.com/user-attachments/assets/cb41f4d2-bed8-4853-a9c1-86906c25d4a4" />

Reply:
From my casual (maybe wrong?) reading of airbyte-ci, it:
1) runs unit tests
2) builds the connector
3) runs integrations and acceptance tests. 

[This](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/java_connectors.py#L103-L140) is where I'm getting the above from.

Reply:
@aaronsteers  we need to call the `integrationTestJava` gradle task. Check doesn't cover this.

Thought: I'm also happy to take it from here. If you merge this in I can modify the workflow to bring it up to parity with what we have now. The current interface is GHA and Poe, which is fine by me.